Model/Config/Source/HouseNumberStreetLine.php

```diff
@@ -0,0 +1,43 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2021 Adyen NV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Model\Config\Source;
+
+use Magento\Framework\Data\OptionSourceInterface;
+
+class HouseNumberStreetLine implements OptionSourceInterface
+{
+    /**
+     * @return array
+     */
+    public function toOptionArray(): array
+    {
+        return [
+            ['value' => 0, 'label' => __('Draw from last street line')],
```

Comment:
I think it would make sense to also add the use default address parser which would just fall back to the current logic. (also to keep the backwards compatibility so we can introduce this still in v6)

Reply:
value=0 'Draw from last street line' is [using the default parser](https://github.com/Adyen/adyen-magento2/pull/977/files#diff-b74fba883fda9c3eff5c703a99610a2c6920b89e1f40dc6bb6f7e7c1d0e6af82R58) that was being used. And in config.xml this value is the default so that instances continue seeing the same results. Or do you mean something else?

Reply:
sorry I thought this setting will just use the last enabled street line as house number or name but not actually use the parse. In case this  is the one falling back to the parser then it is indeed already working correctly :)

Reply:
Indeed, the 'Draw from last street line' option uses the parser